File: Baekjoon/stack/10828.py
import sys
N = int(input())
stack = []

for _ in range(N):
    line = sys.stdin.readline().split()

    if line[0] == "push":
        stack.append(int(line[1]))

    elif line[0] == "pop":
        if len(stack)>0:
            print(stack[-1])
            stack.pop()
        else:
            print(-1)

    elif line[0] == "size":
        print(len(stack))

    elif line[0] == "empty":
        if len(stack) == 0:
            print(1)
        else:
            print(0)

    elif line[0] == "top":
        if len(stack) > 0:
            print(stack[-1])
        else:
            print(-1)

File: Baekjoon/stack/10828.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;
stack <int>STACK;

int main(){
    int n;
    cin>>n;
    for(int i=0;i<n;i++){
        string str="";
        cin>>str;
        if (str == "push") {   
            int x; 
            cin >> x;
            STACK.push(x);
        }
        else if (str == "pop") {
            if (S.empty())
                cout << "-1" << "\n";
            else {
                cout << S.top() << "\n";
                STACK.pop();
            }
        }
        else if(str == "size") {
            cout << STACK.size() << "\n";
        }
     
        
        else if (str == "empty") {
            cout << STACK.empty() << "\n";
        }
        
        else{
            if (STACK.empty())
                cout << "-1"<<"\n";
            else 
                cout << STACK.top() << "\n";
        }
        
    }
    return 0;
}
